collector/zfs.go

```diff
@@ -0,0 +1,95 @@
+package collector
+
+// +build linux freebsd
+// +build !nozfs
+
+import (
+	"errors"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+type zfsMetricValue int
+
+const zfsErrorValue = zfsMetricValue(-1)
+
+var zfsNotAvailableError = errors.New("ZFS / ZFS statistics are not available")
+
+type zfsSysctl string
+type zfsSubsystemName string
+
+const (
+	arc = zfsSubsystemName("zfs_arc")
```

Comment:
All the collectors have a shared namespace, so to be safe this should be `zfsArc`. I'd also just keep this as a plain string.